src/Libraries/CoreNodeModelsWpf/Controls/CustomSelectionControl.xaml.cs

```diff
@@ -0,0 +1,80 @@
+using System.Linq;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Input;
+
+using CoreNodeModels.Input;
+
+using Dynamo.Utilities;
+
+namespace CoreNodeModelsWpf.Controls
+{
+    /// <summary>
+    /// Interaction logic for DropDownControl.xaml
+    /// </summary>
+    public partial class CustomSelectionControl : UserControl
+    {
+        private readonly CustomSelection model;
+
+        /// <summary>
+        /// Create the control for the custom dropdown menu and editor
+        /// </summary>
+        public CustomSelectionControl(CustomSelectionViewModel viewModel)
+        {
+            DataContext = viewModel;
+            model = viewModel.Model;
+
+            InitializeComponent();
+        }
+
+        public ComboBox BaseComboBox { get; set; }
+
+        private void EnumItemsListbox_PreviewKeyDown(object sender, KeyEventArgs e)
+        {
+            if (!( sender is ListBox listBox ))
+            {
+                return;
+            }
+
+            if (e.Key == Key.Tab)
+            {
+                int offset = e.KeyboardDevice.Modifiers == System.Windows.Input.ModifierKeys.Shift ? 0 : 1;
```

Comment:
I am usually not a stickler for style stuff besides `if(){ }` - but the whitespace seems a bit confusing to the eye.